pkg/primitives/vm/utils.go

```diff
@@ -459,3 +480,10 @@ func mergeEnvs(base, new map[string]string) map[string]string {
 
 	return base
 }
+
+func (p *Manager) volumeExist(volName string) bool {
+	ctx := context.Background()
+	storage := stubs.NewStorageModuleStub(p.zbus)
+	_, err := storage.VolumeLookup(ctx, volName)
+	return err == nil
+}
```

Comment:
Why this is a separate function, i see this doesn't make it better because:
- U are using the wrong context? this context since u simply creating a new empty context
- You recreate the storage stubs and u also recreate that in the caller.